roofit/roofitcore/src/RooDataHist.cxx

```diff
@@ -591,7 +591,7 @@ void RooDataHist::importDHistSet(const RooArgList& /*vars*/, RooCategory& indexC
 void RooDataHist::_adjustBinning(RooRealVar &theirVar, const TAxis &axis,
     RooRealVar *ourVar, Int_t *offset)
 {
-  if (!dynamic_cast<RooRealVar*>(ourVar)) {
+  if (!dynamic_cast<RooRealVar*>(static_cast<RooAbsArg*>(ourVar))) {
```

Comment:
This is weird. The interface enforces `RooRealVar*`. If this cast fails, something is deeply broken. What about just checking for `nullptr`?

Reply:
The callers of this function do not do type checking of the argument, but static_cast the argument to RooRealVar*. The type check is then supposed to be done in the function itself. But for the dynamic_cast to actually do the type check the pointer must first be static_cast back to RooAbsArg*, otherwise the compiler will optimise this to a simple nullptr check since the only way a dynamic_cast of a pointer to a pointer of the same type can fail is if it is the null pointer. The code as it was written was expecting the dynamic_cast to fail for a non-null pointer that originally was not a RooRealVar* (even though it had been static_cast to a RooRealVar* before calling the function). Just checking for nullptr does not guarantee that the pointer is actually a RooRealVar*.
This was the minimal change to make the code do what was intended.
The alternative would be to change the signature of the function to use a RooAbsArg*, which avoids casting the pointer to call the function and then casting it back in the function to do the type check. But that is a bigger change, and I proposed the minimal change to fix the warning.

Reply:
I am not sure I understand the point here. If this function is called with `ourVar` not being a `RooRealVar *` then there is a problem somewhere else in the code. 
This function is an internal one and when used one passes always a RooRealVar pointer, so I don;t see any problem. 
A better alternative is to remove the check in any case and pass a reference to RooRealVar. 
I would really avoid doing this double cast